website/guides/lyc_timing.md

```diff
@@ -14,318 +14,742 @@ Written by [Ron Nelson](https://github.com/rondnelson99/)
+                    return { cycles:  4, instr: 'push', fixed: true };
+                case 'reti':
+                    return { cycles:  4, instr: 'reti', legend: 'Return from interrupt', fixed: true };
+                case 'set':
+                    return { cycles:  2, instr: 'set', fixed: false };
+
+                case 'skip':
+                    let cycles = props.cycles;
+                    if (typeof cycles !== 'number' || !isFinite(cycles)) {
+                        throw new SyntaxError('"Skip" CPU op requires a numeric cycle count');
+                    }
+                    return { cycles, class: 'skip', class: 'op' };
+
+                default:
+                    throw new SyntaxError(`Unknown instruction type "${props.op}"`);
+            };
+        })();
+
+        // If a line was requested, override the instruction with it
+        if (props.instr) {
+            info.instr = props.instr;
+        }
+
+        // Immediate instructions and taken jumps take one extra cycle
+        // FIXME: Normally Vue would correctly type these as Booleans as per `props` above, but it doesn't...
+        if (props.immediate !== undefined) {
+            if (info.fixed || info.jump) {
+                throw new SyntaxError(`CPU op ${props.op} cannot be immediate!`)
+            }
+            info.cycles++;
+        }
+        if (props.taken !== undefined) {
+            if (!info.jump) {
+                throw new SyntaxError(`CPU op ${props.op} is not a jump, it cannot be taken!`);
+            }
+            info.cycles++;
+        }
+
+        // Override the legend if explicitly provided
+        if (props.legend) {
+            info.legend = props.legend;
+        }
+        // If the operation has a legend, add its name as a class
+        if (info.legend) {
+            info.class = props.op;
+        }
+        // Override the class if explicitly provided
+        if (props.class) {
+            info.class = props.class;
+        }
+
+        return info;
+    },
+};
+
+export default {
+    components: {
+        Timeline,
+        CPUOp,
+    }
+}
+</script>
+<style>
+.timeline-figure {
+    display: flex;
+    flex-flow: row-reverse wrap;
+    align-items: center;
+    justify-content: space-around;
+
+    --ppu-mode2: #ffd96680;
+    --ppu-mode3: #ff800080;
+    --ppu-mode0: #97d07780;
+}
+
+.timeline tr { background-color: transparent; }
+.timeline td { padding-top: 0; padding-bottom: 0; }
+
+.ppu-mode2 { background-color: var(--ppu-mode2); }
+.ppu-mode3 { background-color: var(--ppu-mode3); }
+.ppu-mode0 { background-color: var(--ppu-mode0); }
+
+.cpu-op         { background-color: #00ffff80; }
+.cpu-access     { background-color: #80808080; }
+.cpu-call       { background-color: #00ff0080; }
+.cpu-critical   { background-color: #ff000080; }
+.cpu-interrupt  { background-color: #ffff0080; }
+.cpu-io-3cycle  { background-image: linear-gradient(to top, #80808080, #80808080 33%, #0000ff80 33%, #0000ff80); }
```

Comment:
```suggestion
.cpu-io-3cycle  { background-image: linear-gradient(to top, #80808080 33%, #0000ff80 33%); }
```
This might fix the rendering bug @avivace reported. I learned more about how CSS gradients work after writing this, so I think the two "edge" colors are redundant here—needs testing.